unit-tests-java8/src/test/java/org/eclipse/collections/test/list/mutable/UnmodifiableListTest.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2022 Goldman Sachs and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.test.list.mutable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.collections.impl.test.junit.Java8Runner;
+import org.eclipse.collections.test.list.UnmodifiableListTestCase;
+import org.junit.runner.RunWith;
+
+@RunWith(Java8Runner.class)
+public class UnmodifiableListTest
+        implements UnmodifiableListTestCase
+{
+    @SafeVarargs
+    @Override
+    public final <T> List<T> newWith(T... elements)
+    {
+        List<T> result = new ArrayList<>(Arrays.asList(elements));
+        return Collections.unmodifiableList(result);
+    }
```

Comment:
Here we show that Collections.unmodifiableList() and our unmodifiable lists pass the same assertions and have the same contract. I wasn't sure this would be true for subtle interactions discussed above, like throwing unsupported vs index out of bound, or throwing vs ignoring when attempting to perform mutations that don't mutate like clearing an empty list.